app/src/main/java/com/appunite/loudius/di/NetworkModule.kt

```diff
@@ -17,94 +17,66 @@
 import io.ktor.http.ContentType
 import io.ktor.serialization.gson.GsonConverter
 import okhttp3.logging.HttpLoggingInterceptor
+import org.koin.core.module.dsl.singleOf
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
 import java.time.LocalDateTime
-import javax.inject.Singleton
 
-@InstallIn(SingletonComponent::class)
-@Module
-object NetworkModule {
-    @Provides
-    @Singleton
-    fun provideLoggingInterceptor(): HttpLoggingInterceptor = HttpLoggingInterceptor().apply {
-        level = HttpLoggingInterceptor.Level.BASIC
+val networkModule = module {
+    single<HttpLoggingInterceptor> {
+        HttpLoggingInterceptor().apply {
+            level = HttpLoggingInterceptor.Level.BASIC
+        }
     }
 
-    @Provides
-    @AuthAPI
-    fun provideBaseAuthUrl() = Constants.AUTH_API_URL
-
-    @Provides
-    @BaseAPI
-    fun provideBaseAPIUrl() = Constants.BASE_API_URL
-
-    @Provides
-    @Singleton
-    @AuthAPI
-    fun provideAuthHttpClient(
-        gson: Gson,
-        @AuthAPI baseUrl: String,
-        loggingInterceptor: HttpLoggingInterceptor,
-    ): HttpClient = HttpClient(OkHttp) {
-        expectSuccess = true
-        engine {
-            addInterceptor(TestInterceptor)
-            addInterceptor(loggingInterceptor)
-        }
-        defaultRequest {
-            url(baseUrl)
-        }
-        install(ContentNegotiation) {
-            register(ContentType.Application.Json, GsonConverter(gson))
+    single<HttpClient>(named("auth")) {
+        HttpClient(OkHttp) {
+            expectSuccess = true
+            engine {
+                addInterceptor(TestInterceptor)
+                addInterceptor(get<HttpLoggingInterceptor>())
+            }
+            defaultRequest {
+                url(Constants.AUTH_API_URL)
+            }
+            install(ContentNegotiation) {
+                register(ContentType.Application.Json, GsonConverter(get()))
+            }
         }
     }
 
-    @Provides
-    @Singleton
-    @BaseAPI
-    fun provideBaseHttpClient(
-        gson: Gson,
-        @BaseAPI baseUrl: String,
-        loggingInterceptor: HttpLoggingInterceptor,
-        authInterceptor: AuthInterceptor,
-        authFailureHandler: AuthFailureHandler,
-    ): HttpClient = HttpClient(OkHttp) {
-        expectSuccess = true
-        engine {
-            addInterceptor(authInterceptor)
-            addInterceptor(TestInterceptor)
-            addInterceptor(AuthFailureInterceptor(authFailureHandler))
-            addInterceptor(loggingInterceptor)
-        }
-        defaultRequest {
-            url(baseUrl)
-        }
-        install(ContentNegotiation) {
-            register(ContentType.Application.Json, GsonConverter(gson))
+    single<HttpClient>(named("base")) {
+        HttpClient(OkHttp) {
+            expectSuccess = true
+            engine {
+                addInterceptor(TestInterceptor)
+                addInterceptor(get<HttpLoggingInterceptor>())
```

Comment:
Question: Why did you delete `authInterceptor` and `authFailureHandler` ?

Reply:
thanks for the sharp eye 🦅 resvoled